static/js/components/forms/CouponDeactivateForm.js

```diff
@@ -0,0 +1,55 @@
+// @flow
+import React from "react";
+import { Formik, Field, Form, ErrorMessage } from "formik";
+import * as yup from "yup";
+
+import FormError from "../../components/forms/elements/FormError";
+
+type CouponDeactvateFormProps = {
+  onSubmit: Function,
+};
+
+const couponValidations = yup.object().shape({
+  coupons: yup
+    .string()
+    .required("At least one coupon name is required")
+    .matches(/^[\w\n]+$/, "Only letters, numbers, and underscores allowed"),
+});
+
+export const CouponDeactivateForm = ({
+  onSubmit,
+}: CouponDeactvateFormProps) => (
+  <Formik
+    onSubmit={onSubmit}
+    validationSchema={couponValidations}
+    initialValues={{
+      coupons: "",
+    }}
+    render={({
+      isSubmitting,
+      setFieldValue,
+      setFieldTouched,
+      errors,
+      touched,
+      values,
+    }) => (
+      <Form className="coupon-form">
+        <div>
+          <div className="block text-area-div">
+            <label htmlFor="coupons">
+              Coupon Names (each coupon at separate line)*
```

Comment:
I am not sure if we discussed this earlier with Bon OR @arslanashraf7. But there are 2 types of coupons. One can have multiple coupon codes against a name. The promo will have a single code.

We should handle both cases. In case, a coupon name has 100's of coupon codes, it will be hard to copy paste all the codes.